framework/di/Container.php

```diff
@@ -364,6 +365,9 @@ protected function build($class, $params, $config)
         }
 
         $dependencies = $this->resolveDependencies($dependencies, $reflection);
+        if (!$reflection->isInstantiable()) {
+            throw new NotInstantiableException($reflection->name);
```

Comment:
It could either throw `InvalidConfigException` or `NotInstantiableException` could be extended from it.